### PR TITLE
Have `service.get_counts()` return `collections.Counter` and `service.run()` return `cirq.Result`

### PR DESCRIPTION
**Pull request by vtomole**

### CODE REVIEW

File: cirq_superstaq/service.py [hunk 1]
```diff
@@ -27,6 +27,56 @@
...
+    counter: collections.Counter, circuit: cirq.Circuit, param_resolver: cirq.ParamResolver
+) -> cirq.Result:
+    """Converts a collections.Counter to a cirq.Result.
+
+    Args:
+            circuit: The circuit to run.
+            repetitions: The number of times to run the circuit.
+            name: An optional name for the created job. Different from the `job_id`.
+            target: Where to run the job. Can be 'qpu' or 'simulator'.
+            param_resolver: A `cirq.ParamResolver` to resolve parameters in  `circuit`.
```
Comment (richrines1) [inline cirq_superstaq/service.py:40]: needs update

File: cirq_superstaq/service.py [hunk 2]
```diff
@@ -27,6 +27,56 @@
...
+
+        Returns:
+            A `cirq.Result` for the given circuit and counter.
+
+    """
+
+    measurement_key_names = list(circuit.all_measurement_keys())
+    measurement_key_names.sort()
+    # Combines all the measurement key names into a string: {'0', '1'} -> "01"
+    combine_key_names = "".join(list(measurement_key_names))
```
Comment (richrines1) [inline cirq_superstaq/service.py:50]: nit
```suggestion
    combine_key_names = "".join(measurement_key_names)
```

Comment (vtomole) [reply cirq_superstaq/service.py]: Nice. Thanks.

**Review by richrines1 — APPROVED** [review]
lgtm!

**Pull request merged**